raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -90,10 +91,11 @@ def handle_block(
         return TransitionResult(payment_state, list())
 
     sub_iteration = initiator.handle_block(
-        payment_state.initiator,
-        state_change,
-        channel_state,
-        pseudo_random_generator,
+        initiator_state=payment_state.initiator,
+        had_canceled_payments=len(payment_state.cancelled_channels) != 0,
```

Comment:
TBH, i don't feel this is semantically correct. A cancelled channel is basically a route which we:
1. Tried but didn't deliver all the way to target.
2. Cancelled manually

The goal is to try to prevent the initiator task from clearing upon expiring the lock with a specific secrethash.

So, i would remove this change and in the initiator's `handle_block` would do as follows:

```
channel.py
def lock_exists(channel_state):
     lock = channel.get_lock(channel_state.our_state)
     if not lock:
          lock = channel.get_lock(channel.partner_state)
     return lock

initiator.py
def handle_block():
..... etc
    return TransitionResult(
        # If there were any refund transfers we need to keep the payment
        # task around to wait for the LockExpired message.
        # Check https://github.com/raiden-network/raiden/issues/3183
        initiator_state if channel.lock_exists(secrethash) else None,
        typing.cast(typing.List[Event], expired_lock_events),
    )	
```

Which basically says: if the lock with the task's secrethash still exists, don't clear the task.

Reply:
> 1. Tried but didn't deliver all the way to target.

So there is a refund transfer involved.

> Cancelled manually

We can't cancel transfers manually right now.

--

Considering the two above is why I went with the current simple solution. But if we ever allow manual canceling of payments then this may break. Let me think about your suggestion.

Reply:
@rakanalh question: in your suggestion you also check for the lock in `our_state`. Would you not only be interested in the partner state for this?

Reply:
I want to check for both, because i make sure that i return that the lock does not exist **IFF** it was removed from both our_state and the partner state. I think calling the function `lock_exists` is a bad idea because it could give the impression that it's generic while it's specific for this scenario... so it is up to you on the naming of it.

Reply:
Okay after our chat in rocketchat I think we are in agreement. For documentation: 

1. Checking locks is better than the length of canceled routes. Doing it with the length of that list will work now but may break in the future as mentioned above.

2. We need to check both states for the lock since we need to make sure that both parties sync. 
    If in the refund scenario above, `B` sends `A` a `RefundTransfer` and later the `LockExpired` before `A` sends `B` a `LockExpired` we will have removed the task and `A` will never get to send `B` a `LockExpired`.